caffe2/core/common_gpu.cc

```diff
@@ -194,7 +194,7 @@ void DeviceQuery(const int device) {
      << std::endl;
   ss << "Total registers per block:     " << prop.regsPerBlock << std::endl;
   ss << "Warp size:                     " << prop.warpSize << std::endl;
-#ifndef __HIPCC__
+#ifndef __HIP_PLATFORM_HCC__
```

Comment:
Oh, is this because we're no longer compiling with hcc?

Reply:
I am a little confused: who is responsible for setting this macro? The documentation suggests that the HIP headers are responsible for defining then. Then it is *definitely* inappropriate to use this macro in a header (because the semantics of the header changes depending on if you included the header before or after the HIP headers), and for good measure, we should probably avoid using it in non-headers as well.

Basically, if you want to add some block of code that is toggled by HIPIFY, it should be some sort of constant that the HIPIFY script replaces with the appropriate new constant; it shouldn't be conditioned on something from a HIP header.

Reply:
we have set this macro in cmake/Dependencies.cmake

Reply:
It's still inappropriate to put `#ifdef` on this macro in headers. Then the header will preprocess differently depending on which compiler invocation calls it (if you build a cc file with or without the flag.)